test/middleware/navigation/check.condition.spec.ts

```diff
@@ -0,0 +1,82 @@
+import { describe, expect, test } from '@jest/globals';
+import {
+  checkBOsOrMOsDetailsNotEntered,
+  checkBeneficialOwnersStatementDetailsNotEntered,
+  checkEntityDetailsNotEntered,
+  checkPresenterDetailsNotEntered,
+  checkHasSoldLand,
+  checkIsSecureRegister
+} from "../../../src/middleware/navigation/check.condition";
+import { BeneficialOwnerGovKey } from '../../../src/model/beneficial.owner.gov.model';
+import { BeneficialOwnerIndividualKey } from '../../../src/model/beneficial.owner.individual.model';
+import { BeneficialOwnerOtherKey } from '../../../src/model/beneficial.owner.other.model';
+import { BeneficialOwnerStatementKey } from '../../../src/model/beneficial.owner.statement.model';
+import { HasSoldLandKey, IsSecureRegisterKey } from '../../../src/model/data.types.model';
+import { EntityKey } from '../../../src/model/entity.model';
+import { ManagingOfficerCorporateKey } from '../../../src/model/managing.officer.corporate.model';
+import { ManagingOfficerKey } from '../../../src/model/managing.officer.model';
+import { PresenterKey } from '../../../src/model/presenter.model';
+import { APPLICATION_DATA_MOCK } from '../../__mocks__/session.mock';
+
+describe("check condition navigation tests", () => {
+
+  test("checkHasSoldLand should return true", () => {
+    const data = checkHasSoldLand({ ...APPLICATION_DATA_MOCK, [HasSoldLandKey]: '1' });
+    expect(data).toEqual(true);
+  });
+
+  test("checkIsSecureRegister should return true", () => {
+    const data = checkIsSecureRegister({ ...APPLICATION_DATA_MOCK, [IsSecureRegisterKey]: '1' });
+    expect(data).toEqual(true);
+  });
+
+  test("checkPresenterDetailsNotEntered should return true", () => {
```

Comment:
Sure, this test is correct but my request is that we add a new test "checkPresenterDetailsNotEntered should return false when presenter data IS present". It would satisfy my curiosity and question on the `check.condition.ts` logic.

Reply:
Should return false if the previous checks pass and obviously, in this context the presenter object is there